common/chat.ts

```diff
@@ -199,3 +241,32 @@ export interface ModerateChannelUserServerRequest {
    */
   moderationReason?: string
 }
+
+/**
+ * Specific chat information about a user in a particular channel, such as their join date, their
+ * channel role (e.g. are they a moderator), etc.
+ */
+export interface ChatUserProfile {
+  joinDate: number
+  /**
+   * User is considered a channel moderator if they have one of the following permissions:
+   *  - owner
+   *  - editPermissions
+   *  - ban
+   *  - kick
+   */
+  isModerator: boolean
+}
+
+export type ChatUserProfileJson = Jsonify<ChatUserProfile>
```

Comment:
If you're going to make a Jsonify'd structure, you should basically always *also* be writing the the `toFooJson` method as you do so. Don't make the user of this type write the conversion every time, it's extremely error-prone